llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -711,11 +689,6 @@ static void LowerEsimdConstructs(Module &M) {
     MPM.add(createEarlyCSEPass(true));
     MPM.add(createInstructionCombiningPass());
     MPM.add(createDeadCodeEliminationPass());
-    MPM.add(createFunctionInliningPassHelper());
```

Comment:
@cmc-rep, I removed the chain of `-inline -sroa -early-cse -instcombine -dce` passes since `-always-inline` should be handling all the inlining opportunities earlier. Please confirm if you think it is safe to do so. All the ESIMD tests pass in my local testing.

Reply:
I am fine with this as long as the change can pass all tests

Reply:
This may also have code quality impact. Would it be possible to have a fall back option, or double check assembly difference before/after the change?

Reply:
We discussed that issue with @kychendev, and agreed that I will compare device code for 1-2 ESIMD tests before and after the change, to ensure that code quality didn't degrade.

Reply:
@kychendev I compared device code for two ESIMD tests.
- Device code for matrix_trasnspose test was identical.
- Device code for BitonkiSort required `early-cse` to be identical.
So I returned the cleanup passes and added a TODO comment.

Reply:
Got it. So it may have potential impact on code quality. Thanks for double checking and update.